### ISSUE TITLE
Support using the database for package and symbol storage

### ISSUE THREAD

**Issue by loic-sharma** [loic-sharma/BaGet#556]
### Is your feature request related to a problem? Please describe.
Today customers must use the filesystem, S3, Azure Blob Storage, or Google Cloud Storage to store packages and symbols.

### Describe the solution you'd like
Customers should be able to store their content in the database.

**Comment by juanperezADD** [comment]
Is there any advantage to store the content of the NuGet packages in database versus filesystem?

**Comment by loic-sharma** [comment]
Yes, this is useful for on-prem scenarios that already production grade databases. You can make a database highly available and geo-redundant. You can do fail-overs with database is a data center goes down. Achieving this with a file system or network share is difficult.

StackOverflow does this for their private NuGet server. Azure Artifacts on-prem also supports this.
